src/server/scripts/Outland/TempestKeep/Eye/boss_kaelthas.cpp

```diff
@@ -769,85 +718,110 @@ struct boss_kaelthas : public BossAI
+                return;
+            scheduler.CancelAll();
+            me->AttackStop();
+            me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+            DoCastAOE(SPELL_KAEL_PHASE_TWO, true);
+            DoCastSelf(SPELL_PERMANENT_FEIGN_DEATH, true);
+            _feigning = true;
         }
-        ScheduleTimedEvent(0s, 2s, [&]{
-            DoCastSelf(SPELL_BELLOWING_ROAR);
-        }, 30s, 40s);
-    }
+     }
 
-    void JustDied(Unit* /*killer*/) override
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
     {
-        if (Creature* kael = _instance->GetCreature(DATA_KAELTHAS))
+        if (spell->Id == SPELL_RESURRECTION && caster->GetEntry() == NPC_KAELTHAS)
         {
-            kael->AI()->DoAction(ACTION_PROGRESS_PHASE_CHECK);
-            if (!_hasDied)
+            me->RemoveAurasDueToSpell(SPELL_PERMANENT_FEIGN_DEATH);
+            me->SetStandState(UNIT_STAND_STATE_STAND);
+            scheduler.Schedule(6s, [&](TaskContext /*context*/)
             {
-                Talk(SAY_SANGUINAR_DEATH);
-                DoCastSelf(SPELL_KAEL_PHASE_TWO, true);
-                kael->AI()->DoAction(ACTION_START_CAPERNIAN);
-                _hasDied = true;
-            }
+                _preventDeath = false;
+                _feigning = false;
+                me->RemoveUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+                DoResetThreatList();
+                me->SetInCombatWithZone();
+                me->SetReactState(REACT_AGGRESSIVE);
+                if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+                {
+                    AttackStart(target);
+                }
+                ScheduleEvents();
+            });
         }
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        Talk(SAY_ADVISOR_DEATH);
         scheduler.CancelAll();
+        DoCastAOE(SPELL_KAEL_PHASE_TWO, true);
     }
 
     void UpdateAI(uint32 diff) override
     {
-        if (!UpdateVictim())
+        scheduler.Update(diff);
+
+        if (_feigning)
             return;
 
-        scheduler.Update(diff);
+        if (!UpdateVictim())
+            return;
 
         if (me->HasUnitState(UNIT_STATE_CASTING))
             return;
 
         DoMeleeAttackIfReady();
     }
 private:
-    InstanceScript* _instance;
-    bool _hasDied;
+    bool _preventDeath;
+    bool _feigning;
 };
 
-struct npc_capernian : public ScriptedAI
+struct npc_lord_sanguinar : public advisor_baseAI
 {
-    npc_capernian(Creature* creature) : ScriptedAI(creature) {
-        _instance = creature->GetInstanceScript();
-        scheduler.SetValidator([this]
-        {
-            return !me->HasUnitState(UNIT_STATE_CASTING);
-        });
-    }
+    npc_lord_sanguinar(Creature* creature) : advisor_baseAI(creature) { };
 
-    void Reset() override
-    {
-        scheduler.CancelAll();
-        _hasDied = false;
-        me->SetReactState(REACT_PASSIVE);
-    }
+    void ScheduleEvents() override {}
```

Comment:
This was not implemented here, so Sanguinar is missing his abilities now

Reply:
Thanks, will pr soon